.github/workflows/build-images.yaml

```diff
@@ -25,7 +25,8 @@ permissions:
 
 jobs:
   RustImages:
-    if: ${{ github.event.label.name == 'CICD:build-images' || contains(github.event.pull_request.labels.*.name, 'CICD:build-images') }}
+    # this expression makes sure that it only triggers on pull_request events that have the magic CICD label (or when the commit was pushed to a protected branch).
+    if: ${{ github.event_name != "pull_request" || github.event.label.name == 'CICD:build-images' || contains(github.event.pull_request.labels.*.name, 'CICD:build-images') }}
```

Comment:
wouldn't this trigger on all events that aren't pull request? https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows 

Can we be more explicit with the conditional here? Instead of using negation and chaining ||, can we use the other way around?

Reply:
e.g. github.event_name == "pull_request" .... github.event_name == "push"

Reply:
this will only trigger on pull requests and labeled events due to the `on: ` clause above, but I agree that the expression was a bit hard to read due to negation. I changed `github.event_name != "pull_request"` to `github.event_name != "push"` - is that better?

Reply:
oh I see. the global `on:` clause limits the possible triggers for the below jobs.